src/account/PluginManager2.sol

```diff
@@ -80,22 +83,31 @@ abstract contract PluginManager2 {
             }
         }
 
-        if (isGlobal) {
-            if (_storage.validationData[validationFunction].isGlobal) {
-                revert GlobalValidationAlreadySet(validationFunction);
+        if (validationConfig.isGlobal()) {
+            if (_storage.validationData[validationConfig.functionReference()].isGlobal) {
+                revert GlobalValidationAlreadySet(validationConfig.functionReference());
             }
-            _storage.validationData[validationFunction].isGlobal = true;
+            _storage.validationData[validationConfig.functionReference()].isGlobal = true;
+        }
+
+        if (validationConfig.isSignatureValidation()) {
+            if (_storage.validationData[validationConfig.functionReference()].isSignatureValidation) {
+                revert SignatureValidationAlreadySet(validationConfig.functionReference());
+            }
```

Comment:
Not sure if we should revert here for those two. Why not just set it to the incoming value?

The current scenarios based on the code are the following. The intentions of what we allow or not is not clear.
incoming true + current true-> revert;
incoming true + current false-> set true; => this can be an update
incoming false + current true -> do nothing;
incoming false + current true -> do nothing;
incoming false + current false -> do nothing;

Reply:
Yeah, good call. I was copying the previous pattern for the global flag, but that doesn't make too much sense. I'll update it to set the incoming value.